src/permutat.c

```diff
@@ -1810,25 +1820,29 @@ Obj             QuoIntPerm2 (
         return QUO( opL, opR );
     }
 
+    Obj inv = STOREDINV_PERM(opR);
+    if ( inv != 0)
+      return INTOBJ_INT(IMAGE(img-1, CONST_ADDR_PERM2(inv), DEG_PERM2(inv))+1);
+    
     /* compute the preimage                                                */
-    pre = img;
-    ptR = ADDR_PERM2(opR);
     if ( img <= DEG_PERM2(opR) ) {
-        while ( ptR[ pre-1 ] != img-1 )
-            pre = ptR[ pre-1 ] + 1;
-    }
-
-    /* return it                                                           */
-    return INTOBJ_INT(pre);
+      pre = (UInt2) img-1;
+      ptR = CONST_ADDR_PERM2(opR);
```

Comment:
Code in this function before your new code uses 4 spaces for indent -> match that?